### PR TITLE
Ensure Avro schema key and value are present as options in mapping [HZ-2757]

### PR DESCRIPTION
**Pull request by Fly-Style**
Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlAvroTest.java [hunk 1]
```diff
@@ -338,6 +343,46 @@ public void test_nonInclusiveUnion() {
...
+        String name = createRandomTopic();
+        kafkaMapping(name, ID_SCHEMA, NAME_SCHEMA)
+                .fields("id BIGINT EXTERNAL NAME \"__key.id\"", "name VARCHAR")
+                .create();
+
+        String query = "SELECT * FROM information_schema.mappings";
+        try (SqlResult result = sqlService.execute(query)) {
+            SqlRow row = null;
+            int cnt = 0;
+            for (SqlRow r : result) {
```
Comment (k-jamroz) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlAvroTest.java:358]: `result.stream().findAny().orElseThrow()`?

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlAvroTest.java [hunk 2]
```diff
@@ -338,6 +343,46 @@ public void test_nonInclusiveUnion() {
...
+                row = r;
+            }
+
+            assertEquals(1, cnt);
+            assertNotNull(row);
+
+            // Destructure the row: it must contain "catalog", "schema", "name", "external name", "type", "options"
+            assertEquals("hazelcast", row.getObject(0));
+            assertEquals("public", row.getObject(1));
+            assertEquals(name, row.getObject(2));
```
Comment (k-jamroz) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlAvroTest.java:369]: these assertions test more infoschema in general than schema options. They are fine, but redundant here, as this should be covered elsewhere (is it?)

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlAvroTest.java [hunk 3]
```diff
@@ -338,6 +343,46 @@ public void test_nonInclusiveUnion() {
...
+            String options = row.getObject(5);
+            assertNotNull(options);
+            // Ensure that the options contain avro schema key and value
+            assertThat(options)
+                    .contains("keyFormat\":\"avro\"")
+                    .contains("valueFormat\":\"avro\"")
+                    .contains("keyAvroSchema\":")
+                    .contains("valueAvroSchema\":");
+        } catch (Exception e) {
+            fail("Failed to execute query: " + query + ". " + e.getMessage());
```
Comment (k-jamroz) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlAvroTest.java:382]: easier to propagate the exception from the test method

**Review by k-jamroz — APPROVED** [review]
LGTM, however worth addressing the comments

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlAvroTest.java (hunk 2 above)
Comment (Fly-Style) [reply hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlAvroTest.java:369]: They are covered as a part of `SqlInfoSchemaTest`. The first couple of schema assertions may seem redundant, but let them be for whole picture understanding.

**Pull request merged**